### PR TITLE
migration: Add case about kill libvirt daemon

### PR DESCRIPTION
**Pull request by cliping**
VIRT-294749 - VM live migration - kill libvirt daemon during PerformPhase of migration

Signed-off-by: cliping <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 4b08d35a078639232f0810f59b8746f5619a31b6
JOB LOG    : /var/lib/avocado/job-results/job-2022-09-27T08.36-4b08d35/job.log
 (01/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_src_virtqemud.with_precopy.p2p: PASS (238.32 s)
 (02/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_src_virtqemud.with_precopy.non_p2p: PASS (238.85 s)
 (03/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_dest_virtqemud.with_precopy.p2p: PASS (243.68 s)
 (04/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_dest_virtqemud.with_precopy.non_p2p: PASS (243.60 s)
 (05/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_dest_virtproxyd.with_precopy.p2p: PASS (241.87 s)
 (06/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_dest_virtproxyd.with_precopy.non_p2p: PASS (242.58 s)
 (07/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_src_libvirtd.with_precopy.p2p: PASS (240.59 s)
 (08/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_src_libvirtd.with_precopy.non_p2p: PASS (240.33 s)
 (09/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_dest_libvirtd.with_precopy.p2p: PASS (242.78 s)
 (10/10) type_specific.io-github-autotest-libvirt.migration.destructive_operations_around_live_migration.migration_kill_libvirt_daemon.no_option.kill_dest_libvirtd.with_precopy.non_p2p: PASS (243.03 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-09-27T08.36-4b08d35/results.html
JOB TIME   : 2416.69 s

File: libvirt/tests/cfg/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.cfg [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    migrate_desturi_type = "tcp"
+    virsh_migrate_desturi = "qemu+tcp://${migrate_dest_host}/system"
+    action_during_mig = '[{"func": "kill_libvirt_daemon", "after_event": "iteration: \'1\'", "func_param": "params"}]'
+    migrate_again_status_error = 'no'
+    migrate_again = 'yes'
+    check_migration_params = '{"xbzrle-cache-size": "67108864", "multifd-channels": "2", "cpu-throttle-initial": "20", "cpu-throttle-increment": "10"}'
+    stress_package = "stress"
+    stress_args = "--cpu 8 --io 4 --vm 2 --vm-bytes 128M --timeout 20s"
+    migrate_speed = "10"
+    #expected_dest_state = "killed"
```
Comment (dzhengfy) [inline libvirt/tests/cfg/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.cfg:36]: Remove this line

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        if expected_dest_state:
+            if not libvirt.check_vm_state(vm.name, expected_dest_state, uri=dest_uri):
+                test.fail("Migrated VM failed to be in %s state at destination." % expected_dest_state)
+        # Check disk
+        backup_uri, vm.connect_uri = vm.connect_uri, dest_uri
+        vm.create_serial_console()
+        vm_session = vm.wait_for_serial_login(timeout=360)
+        vm_session.cmd("echo libvirt_simple_disk_check >> /tmp/libvirt_simple_disk_check")
+        vm_session.close()
+        vm.connect_uri = backup_uri
```
Comment (Yingshun) [inline libvirt/tests/src/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.py:57]: I wonder if you could reuse 'simple_disk_check_after_mig' in post_migration_check()?

File: libvirt/tests/src/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    :param test: test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def setup_zerocopy_option():
+        """
+        Setup for zerocopy_option case
+
+        """
+        test.log.debug("Setup for zerocopy_option case.")
```
Comment (Yingshun) [inline libvirt/tests/src/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.py:26]: test.log.debug should be test.log.info, right?

File: libvirt/tests/cfg/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.cfg (hunk 1 above)
Comment (cliping) [reply libvirt/tests/cfg/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.cfg]: Updated. We need to check domain state on target host. So remove '#'.

File: libvirt/tests/src/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.py (hunk 2 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/destructive_operations_around_live_migration/migration_kill_libvirt_daemon.py:57]: Yes, I agree. It could be reused.

**Comment by cliping** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/3545.

**Review by Yingshun — APPROVED** [review]

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**